src/libprojectM/projectM.cpp

```diff
@@ -565,20 +537,9 @@ void projectM::projectM_init ( int gx, int gy, int fps, int texsize, int width,
 
     initPresetTools(gx, gy);
 
-
 #if USE_THREADS
-
-    #ifdef SYNC_PRESET_SWITCHES
-    pthread_mutex_init(&preset_mutex, NULL);
-#endif
-
-    worker_sync.reset();
-    if (pthread_create(&thread, NULL, thread_callback, this) != 0)
-    {
-
-        std::cerr << "[projectM] failed to allocate a thread! try building with option USE_THREADS turned off" << std::endl;;
-        exit(EXIT_FAILURE);
-    }
+    m_workerSync.Reset();
+    m_workerThread = std::thread(&projectM::ThreadWorker, this);
```

Comment:
We may want to catch `std::system_error` in case the thread could not be started. Calling `exit()` in a library is bad habit, so having a generic exception handler in the C wrapper's `projectm_create*()` functions might be a better option, returning nullptr if construction fails. Then again, throwing in the constructor may cause memory leaks...

Reply:
Throwing in the constructor calls the destructors, so if we use `unique_ptr` for resource management leaking shouldn't be an issue if we throw in the constructor.